rust/agama-cli/src/profile.rs

```diff
@@ -76,123 +70,245 @@ pub enum ProfileCommands {
         /// AutoYaST specific ones. Supported files are json, jsonnet, sh for Agama profiles and ERB, XML, and rules/classes directories
         /// for AutoYaST support.
         url: String,
-        /// Specific directory where all processing happens. By default it uses a temporary directory
+        /// Ignored, formerly the directory for intermadiate files.
         dir: Option<PathBuf>,
     },
 }
 
-fn validate(path: &PathBuf) -> anyhow::Result<()> {
-    let validator = ProfileValidator::default_schema()?;
-    let result = validator
-        .validate_file(path)
-        .context(format!("Could not validate the profile {:?}", path))?;
-    match result {
-        ValidationResult::Valid => {
-            println!("{} The profile is valid.", style("\u{2713}").bold().green(),);
+/// TODO better name
+// Represents the ways user can specify the input on the command line
+// and passes appropriate representations to the web API
+enum CliInput {
+    Url(String),
```

Comment:
and same here with `url::Url` as type instead of plain string. String can be used for passing stuff around, but I kind of expect proper data type here.

Reply:
This turns out to be tricky. The **clap** CLI library does not seem to like it when argument parsing can fail.

Reply:
interesting. I kind of expect that clap can do nice error report when you pass invalid URL.

Reply:
Ah, here's an example of how to do value errors, https://docs.rs/clap/latest/clap/_cookbook/typed_derive/index.html